src/vm_cpu.rs

```diff
@@ -408,6 +408,8 @@ impl VmCpuArchState {
         scounteren.modify(scounteren::instret.val(1));
         regs.guest_regs.scounteren = scounteren.get();
 
+        regs.vs_csrs.htimedelta = htimedelta;
```

Comment:
Instead of setting the htimedelta in VmCpuArchState during creation, define an helper function to set it ?
We don't know how migration APIs will look like right now but htimedelta needs to be updated at that time.
We may or may not need to create VmCpuArchState during that time. 

If you like to keep it this way, leave a TODO comment at least.